### PR TITLE
Intly 1637 add pr status

### PR DESCRIPTION
**Pull request by mikenairn**
## What
* Add Integeration test stage to ga branch job.
* Add PR status updates around integration test stage. 
* Add "tower_environment: dev" as extra vars to all tower jobs.

### CODE REVIEW

**Comment by mikenairn** [comment]
This will trigger tests on change to a ga branch if the "integration tests" label is applied to the associated PR see https://github.com/integr8ly/installation/pull/536

File: jobs/delorean/gitea/ga/branch.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@
           name: 'productName'
           default: 'gitea'
           description: '[REQUIRED] Product to check, this affects the way the job verifies if a new version is available'
+      - bool:
+          name: 'dryRun'
+          default: false
+          description: '[OPTIONAL][Test] Dry run only, only prints what it would do!'
```
Comment (mikenairn) [inline jobs/delorean/gitea/ga/branch.yaml]: Will add this to all branch jobs. @mikenairn ToDo

File: jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile [hunk 2]
```diff
@@ -107,7 +146,52 @@ node {
...
+        when(sourceChanges && runIntegrationTests) {
+            withCredentials([usernamePassword(
+                    credentialsId: githubUserPassCredentialsID,
+                    passwordVariable: 'GITHUB_PASSWORD',
+                    usernameVariable: 'GITHUB_USERNAME')]) {
+                final GitHub gitHub = new GitHubBuilder()
+                        .withOAuthToken(env.GITHUB_PASSWORD, env.GITHUB_USERNAME)
+                        .build()
+                ghUpdatePrCommitStatus(gitHub, prUrl, GHCommitState.PENDING, env.BUILD_URL, 'Pending', ghCommitStatusContext)
+            }
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: Tried to refactor this, but kept hitting issues with jenkins pipelines and objects that cant be serialized. Ended up with quite a lot of repeated code, but it does at least work.

File: jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile [hunk 3]
```diff
@@ -61,6 +62,38 @@ GHPullRequest ghFindOrCreatePullRequest(GHRepository repo, String head, String b
...
+    String ghOwner = pathSegments[0]
+    String ghRepo = pathSegments[1]
+    String ghPrNumber = pathSegments[2]
+    GHRepository repo = gitHub.getRepository("${ghOwner}/${ghRepo}")
+    return repo.getPullRequest(ghPrNumber as int)
+}
+
+@NonCPS
+GHCommitStatus ghUpdatePrCommitStatus(GHPullRequest pr, GHCommitState state, String targetUrl, String description, String context) {
+    return pr.getRepository().createCommitStatus(pr.getHead().getSha(), state, targetUrl, description, context)
```
Comment (JameelB) [inline jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: Do we need to have this in a separate function with the same name? Can we return this on line [83](https://github.com/integr8ly/ci-cd/pull/136/files#diff-19bdd564b8ecac41d9540c5b8ebfa55dR83) instead?

Comment (mikenairn) [reply jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: We could, but this was copied form an external lib that we will more than likely just end up using, so left it as is.

Comment (mikenairn) [reply jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: https://github.com/feedhenry/fh-pipeline-library/blob/master/src/org/feedhenry/GitHubUtils.groovy#L89-L97

**Pull request merged**